.github/workflows/build.yml

```diff
@@ -0,0 +1,119 @@
+name: Build
+
+on:
+  push:
+    branches:
+      - master
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - master
+  schedule:
+    - cron: '0 4 * * *' # Runs every day at 4am: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#scheduled-events-schedule
+
+jobs:
+
+  build:
+    name: ${{ matrix.os }}, Node.js v${{ matrix.node }}, Python ${{ matrix.python }}
+
+    strategy:
+      matrix:
+        os: [windows-2019, ubuntu-18.04, macos-10.15]
+        node: ['12.x']
+        python: ['2.x']
+        include:
+          - os: ubuntu-18.04
+            node: '12.x'
+            python: '3.x'
+            tests: 'skip'
+
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 60
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0 # To fetch all history for all branches and tags. (Will be required for caching with lerna: https://github.com/markuplint/markuplint/pull/111)
+
+      - name: Use Node.js ${{ matrix.node }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node }}
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Use Python ${{ matrix.python }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+
+      - name: Build
+        shell: bash
+        run: |
+          yarn --skip-integrity-check --network-timeout 100000
+          npx electron-replace-ffmpeg
+          npx electron-codecs-test
+          ./scripts/check_git_status.sh
+        env:
+          NODE_OPTIONS: --max_old_space_size=4096
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # https://github.com/microsoft/vscode-ripgrep/issues/9
+
+      - name: Test (headless)
+        if: matrix.tests != 'skip'
+        shell: bash
+        run: |
+          yarn test:theia
+
+      - name: Test (browser)
+        if: matrix.tests != 'skip' && runner.os == 'Linux'
+        uses: GabrielBB/xvfb-action@v1
+        with:
+          run: yarn test:browser
+
+      - name: Test (electron)
+        if: matrix.tests != 'skip' && runner.os == 'Linux'
+        uses: GabrielBB/xvfb-action@v1
+        with:
+          run: yarn test:electron
+
+  publish:
```

Comment:
That might be for the best because as it is written right now the publish job redoes what the build job does...

Reply:
> That might be for the best because as it is written right now the publish job redoes what the build job does...


> Could we just split this workflow in 2 ?

This PR has no intention of making the build smarter or more stable than we have now. Feel free to open up a follow-up once this is merged.

Reply:
> * one workflow that apply on master branch and build from master branch but running only on linux for example ?

@benoitf this is what I went for in my PR but it turns out that it is more "sure" to test all platforms again on publish:
- The workflow for PRs usually test the incoming branch only.
- The workflow for merged commit will test the latest state for master with the new changes.
- GitHub only performs git conflict checks on PR branches, when merged it can still change the behavior of apps...

Correct me if I am wrong.